src/v2i-hub/ODEForwardPlugin/src/ODEForwardPlugin.cpp

```diff
@@ -137,28 +165,65 @@
   * @param msg BSMMessage that is received
   * @routeable_message not used
   */
- void ODEForwardPlugin::HandleRealTimePublish(BsmMessage &msg,
- 		routeable_message &routeableMsg) {
+ void ODEForwardPlugin::HandleRealTimePublish(BsmMessage &msg, routeable_message &routeableMsg) {
+	if(_communicationModeHelper->getUDPMode()){
+		sendUDPMessage(routeableMsg, UDPMessageType::BSM);
+	}else if(_communicationModeHelper->getKafkaMode()){
+		sendBsmKafkaMessage(msg, routeableMsg);
+	}else{
+		PLOG(logERROR) << "Unknown communication mode: " << _communicationMode;
+	}
+ }
+
+ void ODEForwardPlugin::HandleSPaTPublish(SpatMessage &msg, routeable_message &routeableMsg) {
+	if(_communicationModeHelper->getUDPMode()){
+		sendUDPMessage(routeableMsg, UDPMessageType::SPAT);
+	}else if(_communicationModeHelper->getKafkaMode()){
+		sendSpatKafkaMessage(msg, routeableMsg);
+	}else{
+		PLOG(logERROR) << "Unknown communication mode: " << _communicationMode;
+	}
+ }
 
+ void ODEForwardPlugin::HandleTimPublish(TimMessage &msg, routeable_message &routeableMsg) {
+	if(_communicationModeHelper->getUDPMode()){
+		sendUDPMessage(routeableMsg, UDPMessageType::TIM);
+	}else{
+		PLOG(logERROR) << "Unknown communication mode: " << _communicationMode;
+	}
+ }
+
+
+ void ODEForwardPlugin::HandleMapPublish(MapDataMessage &msg, routeable_message &routeableMsg) {
+	if(_communicationModeHelper->getUDPMode()){
+		sendUDPMessage(routeableMsg, UDPMessageType::MAP);
+	}else{
+		PLOG(logERROR) << "Unknown communication mode: " << _communicationMode;
+	}
```

Comment:
This log statement is maybe confusing. Maybe we should updated to unsupported communication mode for message